src/ingestion/kafka-queue.ts

```diff
@@ -71,7 +102,7 @@ export class KafkaQueue implements Queue {
             }
             const singleIngestionTimer = new Date()
             await this.saveEvent(event)
-            resolveOffset(event.kafka_offset!)
+            resolveOffset(eventUuidToKafkaOffset.get(event.uuid!)!)
```

Comment:
TODO: don't run this if the event was not present in batch from Kafka (meaning it was inserted by a plugin's `processEventBatch`)